analyzers/src/SonarAnalyzer.Common/Helpers/SonarLintXmlReader.cs

```diff
@@ -28,16 +28,15 @@ namespace SonarAnalyzer.Helpers;
 
 public class SonarLintXmlReader
 {
-    public static readonly SonarLintXmlReader Empty = new(null, LanguageNames.CSharp);
+    public static readonly SonarLintXmlReader Empty = new(null);
 
     private readonly SonarLintXml sonarLintXml;
-    private readonly string propertyLanguage;
 
     private bool? ignoreHeaderComments;
-    public bool IgnoreHeaderComments => ignoreHeaderComments ??= ReadBoolean(ReadSettingsProperty($"sonar.{propertyLanguage}.ignoreHeaderComments"));
+    public bool IgnoreHeaderComments(string language = LanguageNames.CSharp) => ignoreHeaderComments ??= ReadBoolean(ReadSettingsProperty($"sonar.{LanguageXmlName(language)}.ignoreHeaderComments"));
```

Comment:
Let's add UT that has C# `true` and VB `false`. 
The UT should ask value for C#. And then for VB.
I don't think it's going to work as expected now

Reply:
@pavel-mikula-sonarsource I'm not sure I understand the setup.
What should the sonarlint.xml have? 
the Property `sonar.{LanguageXmlName(language)}.ignoreHeaderComments` for both languages for example?

Reply:
To my knowledge, each SonarLint.xml file should have both properties with the same language. They are actually 4 (`ignoreHeaderComments`, `file.suffixes`, `roslyn.ignoreIssues`, `analyzeGeneratedCode`), but we use only two. The scanner is creating two distinct SonarLint.xml file in two different paths differentiated by language:
- `...\.sonarqube\conf\cs\SonarLint.xml`
- `...\.sonarqube\conf\vbnet\SonarLint.xml`

Reply:
Yes, values for both. And the reader should be able to read the correct value.

While in practice, it will most likely not happen that a single reader would be asked for both values during the existence of the instance, this prerequisite is too far away from this class code.

The behavior of this class should be predictable.

Reply:
To implement this, there should be something like
```
private bool? ignoreHeaderCommentsCS;
private bool? ignoreHeaderCommentsVB;
    public bool IgnoreHeaderComments(string language) =>
        language switch {
LanguageNames.CSharp => ignoreHeaderCommentsCS ??= ReadBoolean(ReadSettingsProperty("sonar.cs.ignoreHeaderComments")),
LanguageNames.CSharp => ignoreHeaderCommentsVB ??= ReadBoolean(ReadSettingsProperty("sonar.vbnet.ignoreHeaderComments")),
_ => throw new UnexpectedLanguageException(language)
}
   
```
where you can add a constructor overload to that exception

Reply:
..and this will get futher simplified in the other class refactoring about its inner state in another PR

Reply:
For general awareness: `SonarLint.xml` can be generated by SonarLint extension, S4NET or it can be crafted manually by users. And users can share the file between their C# and VB projects in the solution.